tmt/utils.py

```diff
@@ -290,6 +290,91 @@ def get_output(self) -> str:
 
 CommonDerivedType = TypeVar('CommonDerivedType', bound='Common')
 
+# TODO: once mypy learns to verify LiteralString, we should really, really enforce
+# the type of each and every command element. It would also make the analysis stronger
+# as it would be able to detect places where text blobs containing shell scripts are
+# injected into commands that would not be handled executed under shell.
+#
+# So, something like this would be one of the options.
+#
+# QuotedString = NewType('QuotedString', str)
+# ShellScript = NewType("ShellScript", str)
+#
+# Commands - executed without `shell=True`/`bash -c ...` - consist of either literal strings
+# or strings:
+#
+#   CommandElement = Union[LiteralString, str]
+#   Command = List[CommandElement]
+#
+# Commands to be execuetd with `shell=True`/`bash -c` can consist of literal strings or
+# correctly escaped strings:
+#
+#   ShellCommandElement = Union[LiteralString, QuotedString]
+#   ShellCommand = List[ShellCommandElement]
+#
+# And we'd have nice helpers to handle conversions (mypy would not let us combine literal strings
+# and values given to tmt via `--script` options & like):
+#
+# def quote_string(s: str) -> QuotedString:
+#     return QuotedString(shlex.quote(s))
+#
+# Bad:  run(['cat'] + [             path  for path in self.opt('paths-to-list')], shell=True)
+# Good: run(['cat'] + [quote_string(path) for path in self.opt('paths-to-list')], shell=True)
+#
+# Until that, at least make a distinction between a command-like and shell-like strings.
+ShellScript = NewType('ShellScript', str)
+_CommandElement = str
+Command = List[_CommandElement]
+
+
+def command_to_command_element(command: Command) -> _CommandElement:
+    """
+    Convert a command to a shell command line element.
+
+    Use when a command or just a list of command options  should become a part
```

Comment:
```suggestion
    Use when a command or just a list of command options should become a part
```

Reply:
Fixed in 1f8d30d7376c4df700417388c0776e0f85cdc88d.